trainer.py

```diff
@@ -30,28 +31,30 @@ def cli_main(
     save_config_callback=None,
     seed_everything_default=42,
     trainer_defaults: dict = {
-        "accelerator": "auto",
+        "accelerator": "gpu",
+        "devices": 4,
+        "strategy": "ddp",
+        "precision": "bf16-mixed",
+        "log_every_n_steps": 1,
+        "max_epochs": 100,
+        "accumulate_grad_batches": 5,
         "callbacks": [
             ModelCheckpoint(
-                # dirpath="checkpoints/",
+                dirpath="checkpoints/",
                 auto_insert_metric_name=False,
                 filename="mae_epoch-{epoch:02d}_val-loss-{val/loss:.2f}",
                 monitor="val/loss",
                 mode="min",
                 save_last=True,
-                save_top_k=3,
+                save_top_k=2,
                 save_weights_only=True,
                 verbose=True,
             ),
-            # LearningRateMonitor(logging_interval="step"),
-            # LogIntermediatePredictions(),
+            LearningRateMonitor(logging_interval="step"),
+            LogIntermediatePredictions(),
         ],
-        "logger": False,  # WandbLogger(project="CLAY-v0", log_model=False)
+        "logger": [WandbLogger(project="CLAY-v0", log_model=False)],
```

Comment:
Ok to add these loggers for now, but I'd like to disable them by default so that new users don't need to install and setup `wandb` and such on the first run. We could potentially use LightningCLI's YAML config files - https://lightning.ai/docs/pytorch/2.0.2/cli/lightning_cli_advanced.html to store some of these advanced configurations.

Reply:
Agreed

Reply:
Another reason is that turning on the WandbLogger by default here also means `wandb` is invoked during inference/prediction, and we would need to do something like `WANDB_MODE=disabled python trainer.py predict ...` to not use wandb at all.